run.sh

```diff
@@ -1,36 +1,74 @@
-#!/bin/bash
+#!/usr/bin/env bash
 
-waitUrl="`pwd`/ui/waiting-room.html";
+set -e
 
-# Ensure dependencies are installed.
-echo "* Checking dependencies..."
-hash tsc 2>/dev/null
-if [ $? -ne 0 ]; then
-  echo "Please install the typescript compiler with ('sudo npm install -g typescript') before continuing."
-  exit
-fi
-hash multirust 2>/dev/null
-if [ $? -ne 0 ]; then
-  echo "Please install multirust with ('./install-multirust.sh') before continuing."
-  exit
-fi
+waitUrl="$(pwd)/ui/waiting-room.html";
+rustVersion="nightly-2015-08-10"
+debugFlag=false
+noBrowserFlag=false
 
-pushd .
-  # Try using the typescript compiler (tsc) to compile UI
-  echo "* Compiling Editor..."
-  cd ui
+# Parse command line options.
+while test $# -gt 0; do
+case "$1" in
+  -h|--help)
+    echo "Compile and run the Eve editor"
+    echo ""
+    echo "Usage:"
+    echo "    run.sh [options]"
+    echo ""
+    echo "Options:"
+    echo "    -h, --help          Print this message"
+    echo "    -d, --debug         Debug build"
+    echo "    -n, --no-browser    Do not open editor in browser"
+    exit 0
+    ;;
+  -d|--debug)
+    debugFlag=true
+    shift
+    ;;
+  -n|--no-browser)
+    noBrowserFlag=true
+    shift
+    ;;
+  *)
+    break
+    ;;
+esac
+done
+
+# Ensure that dependencies are installed.
+printf "* Checking dependencies..."
+deps="tsc multirust"
+for dep in $deps; do
+  if ! which "$dep" &> /dev/null; then
+    printf "  x Please install %s: " "$dep"
+    if [ "$dep" = "tsc" ]; then
+      echo "sudo npm install -g typescript"
+    elif [ "$dep" = "multirust" ]; then
+      echo "./install-multirust"
+    fi
+    exit 1
+  fi
+done
+echo "done."
 
-  tsc
+# Try using the TypeScript compiler (tsc) to compile UI.
+pushd . &> /dev/null
+  printf "* Compiling editor..."
+  cd ui
+  tscError=$(tsc)
   if [ $? -ne 0 ]; then
```

Comment:
`set -e` means we never get into this branch, and since you capture the output of tsc instead of letting it write to screen, we lose the failure output. I'm not sure what the right solution here is though.